aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp

```diff
@@ -3033,6 +3033,11 @@ c10::MaybeOwned<Tensor> maybe_expand_pivots(const Tensor& b,const Tensor& pivots
 }
 
 static void lu_solve_kernel(const Tensor& LU, const Tensor& pivots, const Tensor& B, TransposeType trans) {
+  // Trivial case. We need it here as it makes some backends fail
+  if (B.numel() == 0) {
+    return;
+  }
```

Comment:
What backends fail?
What happens if LU or pivots have 0 elements?

Reply:
cublas batched says that the inputs are incorrect. On  a second thought, I should lift this check to the main function.

Reply:
fwiw, the test that was failing was the broadcasting one you wrote

Reply:
But this shortcut is fine regardless, right?

Reply:
This shortcut is fine. Why don't we have the same for `LU` and `pivots`? Is it because the assumption is that it can't happen here that `B` has more than 0 elements while LU or pivots have 0?

Lifting this check to the main function is fine or having returns only for specific backend functions that can't handle this input.

Reply:
Lifting it to the main function is the right thing to do, indeed. I'll do that.

And yes, the condition that `B` has zero elements implies that  `LU` and/or` `pivots` have zero elements. That's why it's the only one I check.

Reply:
So, it turns out that I had already had put it in the main function. Now, `lu_solve_stub` is called from many places in the code (UGH). I'm preemptively leaving this check there for now with a TODO comment.

Reply:
> Now, lu_solve_stub is called from many places in the code (UGH).

That's why my preference is to have these returns only close to the backend which doesn't work.

Reply:
My point here is that each stub should be called from the function that implements the preparation of the inputs for the given backend. Any other ATen function that needs calling that backend, should do it via the ATen function, as the overhead for these functions is negligible, and it will get to use all the calling optimisations that that function already implements.

Reply:
We have different opinions about this and it's fine. Usually, the backend functions handle inputs with 0 in the shape just fine removing the need to treat it specially in the code.